pandas/core/arrays/period.py

```diff
@@ -782,7 +785,7 @@ def period_array(
     data = np.asarray(data)
 
     if freq:
-        dtype = PeriodDtype(freq)
+        dtype = PeriodDtype(freq)  # type: Optional[PeriodDtype]
```

Comment:
I think want to refactor this entire condition to look as follows:

```python
dtype = None  # type: Optional[PeriodDtype]
if freq:
    dtype = PeriodDtype(freq)
```

FWIW this is actually called out as one of the rationales for Variable Annotations in PEP 526:

https://www.python.org/dev/peps/pep-0526/#rationale

Since we are still supporting 3.5 though can't necessarily use the syntax described there

Reply:
Actually I think OK to ignore my comment above. We might want to revisit after dropping Py35 support if we can add a variable annotation but don't think worth changing code around now if this works for the type checker

Reply:
can you add a comment here on why this is needed